### ISSUE TITLE
Ejecting WebPack config

### ISSUE THREAD

**Issue by OlegWock** [webextension-toolbox/webextension-toolbox#174]
Thank you for this awesome tool! This is not a bug report, but something like question or maybe feature request. It sometimes required to change build process, sometimes webextension-toolbox-config.js works like a charm, but unfortunately not always. Sometimes I need to change built-in webpack config and I wonder can I somehow do 'eject' (like at create-react-app) and have webpack config in my project folder, so I don't need to monkeypatch code from node_modules nor fork this repo and change config for every project?

So, long story short: can I somehow easily take webextension-toolbox's WebPack config as base and change it according to my needs per project?

I will be happy to submit PR with this feature or at least help with it, but I'm not so good in all this webpack things and don't even know how this should be implemented so will be thankful if you point me in right direction haha

**Comment by CanRau** [comment]
You can add a `webextension-toolbox-config.js` to the root of your project next to package.json
I use this to extend babel

```
const { join } = require('path')
const webpack = require('webpack')

module.exports = {
  webpack: (config, env) => {
    config.module.rules = config.module.rules.map(rule => {
      if (rule.use && rule.use.loader.includes('babel-loader')) {
        rule.use.options = {
          babelrc: true,
          extends: join(__dirname + '/.babelrc'),
          cacheDirectory: true,
          envName: env.dev ? 'development' : 'production',
        }
      }
      return rule
    })
    return config
  },
}
```
This code allows you to use a `.babelrc` but it's actually overriding the default, you could alter it to just merge the two, or make your changes directly inside the above code.
I `console.log(config)` to know what to search for..

Hope that helps?
Uh and I believe that file can do even more besides the webpack property, not sure what is available, I was too lazy to dig deeper^^

**Comment by OlegWock** [comment]
Yep, that's works for most cases, but if I understand correctly, using this file I can't change config for plugins, that are already added to config by webextension-toolbox

**Comment by CanRau** [comment]
hum, my example is modifying the existing babel, that's not what you want?

**Comment by OlegWock** [comment]
It does, but you modify options of Babel loader, which are passed as plain object. Instead I mean plugins which accept options at instance creating, like this 

https://github.com/webextension-toolbox/webextension-toolbox/blob/master/src/webpack-config.js#L123

**Comment by CanRau** [comment]
Uh got it, no idea 😅

**Comment by HaNdTriX** [comment]
**File:** `webextension-toolbox-config.js`

```js
const webpack = require('webpack')

module.exports = {
  webpack: (config, env) => {
    // Remove ProvidePlugin
    config.plugins = config.plugins.filter(plugin => !(plugin instanceof webpack.ProvidePlugin))
    return config
  }
}
```

**Comment by tm1000** [comment]
I have no idea how rejecting works in React because I never use create-react-app. 

My thoughts would be that ejecting would just move the config up and then have it copy itself out.
